Sources/OpenAISwift/OpenAISwift.swift

```diff
@@ -139,6 +141,53 @@ extension OpenAISwift {
             }
         }
     }
+    
+    /// Send a Chat request to the OpenAI API with stream enabled
+    /// - Parameters:
+    ///   - messages: Array of `ChatMessages`
+    ///   - model: The Model to use, the only support model is `gpt-3.5-turbo`
+    ///   - user: A unique identifier representing your end-user, which can help OpenAI to monitor and detect abuse.
+    ///   - temperature: What sampling temperature to use, between 0 and 2. Higher values like 0.8 will make the output more random, while lower values like 0.2 will make it more focused and deterministic. We generally recommend altering this or topProbabilityMass but not both.
+    ///   - topProbabilityMass: The OpenAI api equivalent of the "top_p" parameter. An alternative to sampling with temperature, called nucleus sampling, where the model considers the results of the tokens with top_p probability mass. So 0.1 means only the tokens comprising the top 10% probability mass are considered. We generally recommend altering this or temperature but not both.
+    ///   - choices: How many chat completion choices to generate for each input message.
+    ///   - stop: Up to 4 sequences where the API will stop generating further tokens.
+    ///   - maxTokens: The maximum number of tokens allowed for the generated answer. By default, the number of tokens the model can return will be (4096 - prompt tokens).
+    ///   - presencePenalty: Number between -2.0 and 2.0. Positive values penalize new tokens based on whether they appear in the text so far, increasing the model's likelihood to talk about new topics.
+    ///   - frequencyPenalty: Number between -2.0 and 2.0. Positive values penalize new tokens based on their existing frequency in the text so far, decreasing the model's likelihood to repeat the same line verbatim.
+    ///   - logitBias: Modify the likelihood of specified tokens appearing in the completion. Maps tokens (specified by their token ID in the OpenAI Tokenizer—not English words) to an associated bias value from -100 to 100. Values between -1 and 1 should decrease or increase likelihood of selection; values like -100 or 100 should result in a ban or exclusive selection of the relevant token.
+    ///   - onEventReceived: Called Multiple times, returns an OpenAI Data Model
+    ///   - onComplete: Triggers when sever complete sending the message
+    public func sendChatWithStreamEnabled(with messages: [ChatMessage],
```

Comment:
It might make more sense for this to be called syntactically `sendStreamingChat` or `sendChatWithStreaming`. (A Swift-ification might have an `AsyncSequence` for the output, but that's probably harder to implement)

Reply:
I'd rather use `AsyncStream`, like this : 

```swift
func generateNumbers() async -> AsyncStream<Int> {
    let stream = AsyncStream<Int> { continuation in
        Task.detached {
            for i in 1...10 {
                await continuation.yield(i)
            }
            continuation.finish()
        }
    }
    return stream
}

@MainActor
func printNumbers() async {
    await for number in generateNumbers() {
        print(number)
    }
}
```

Reply:
I saw we keep both closure one and concurrency one in extension, I created wrap function with `AsyncStream` in extension : https://github.com/adamrushy/OpenAISwift/pull/63/commits/dd66aa7e104f07e0ef16677f3739586dcb4027f6

``` swift
for await result in openAI.sendStreamingChat(with: [
    ChatMessage(role: .user, content: "write 1000 words story")
]) {
    switch result {
    case .success(let success):
        print(success.choices?.first?.delta.content ?? "")
    case .failure(let failure):
        print(failure.localizedDescription)
    }
}

// output:
Once
 upon
 a
 time
,
 in
 a
 small
 village
 nestled
 in
...
```